hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieAppendHandle.java

```diff
@@ -249,13 +249,13 @@ private Option<HoodieRecord> prepareRecord(HoodieRecord<T> hoodieRecord) {
 
         // Prepend meta-fields into the record
         MetadataValues metadataValues = populateMetadataFields(finalRecord);
-        HoodieRecord populatedRecord =
+        Option<HoodieRecord> populatedRecord =
```

Comment:
for records that are deleted, isn't finalRecordOpt.isPresent() will be false and hence the condition in L243 will be false and we will not hit this condition only.